### ISSUE TITLE
Restic integration errors out

### ISSUE THREAD

**Issue by trondhindenes** [heptio/ark#567]
I'm trying to get restic up and running but it keeps failing. My ark config looks like this:
```
backupStorageProvider:
  name: aws
  bucket: stuff-dev-heptio-ark
  config:
    region: eu-west-1
  resticLocation: stuff-dev-heptio-ark-restic
```
(btw _please_ provide yaml files instead of fancy kubectl/jq commands to configure the required settings. While those are super-impressive, yaml is easier to read and understand. I have no idea what most of the commands in the restic documentation do, which makes it very hard to follow.

Here's the error I'm getting from the ark server:
```
time="2018-06-21T21:02:16Z" level=error msg="backup failed" error="error running command, stderr=Fatal: create repository at s3:s3.amazonaws.com/stuff-dev-heptio-ark-restic/nginx-example failed: client.BucketExists: The specified bucket does not exist.\n\n: exit status 1" key=heptio-ark/nginx-example logSource="pkg/controller/backup_controller.go:280"
```
Running ark without restic integration works, and the iam permissions are correct. The bucket exists, but maybe the restic integration expects a certain bucket location? AFAIK there's no `resticLocationRegion` setting or similar.

**Comment by skriss** [comment]
Hi @trondhindenes, thanks for testing out the alpha and reporting this.

We will have full YAML sample files for GA, but as of the initial alpha those were still in-flight. Apologies.

When you set up the bucket for restic, did you go through the IAM setup instructions as per https://github.com/heptio/ark/blob/master/docs/aws-config.md#create-iam-user, Step #2? One thing to note here is that I *think* you need to use a different policy name than `heptio-ark` when creating this for the restic bucket, since the `heptio-ark` policy already exists for the main Ark bucket and is  attached to the `heptio-ark` IAM user.

**Comment by trondhindenes** [comment]
I just added the "restic bucket" to the ark policy, it reads like this:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "ec2:DescribeVolumes",
                "ec2:DescribeSnapshots",
                "ec2:CreateTags",
                "ec2:CreateVolume",
                "ec2:CreateSnapshot",
                "ec2:DeleteSnapshot"
            ],
            "Resource": "*"
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:GetObject",
                "s3:DeleteObject",
                "s3:PutObject",
                "s3:AbortMultipartUpload",
                "s3:ListMultipartUploadParts"
            ],
            "Resource": [
                "arn:aws:s3:::stuff-dev-heptio-ark/*",
                "arn:aws:s3:::stuff-dev-heptio-ark-restic/*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:ListBucket"
            ],
            "Resource": [
                "arn:aws:s3:::stuff-dev-heptio-ark",
                "arn:aws:s3:::stuff-dev-heptio-ark-restic"
            ]
        }
    ]
}
```
should be a totally valid iam policy. I've verified that I can upload files to the restic bucket using the ark iam user, so everything should be good on the aws/iam side of things.

That said I totally understand that it's early and that I might have to wait a bit before documentation is finalized, so no worries :-)

**Comment by skriss** [comment]
OK, thanks - let me look into this a bit more and I'll get back to you.  You shouldn't have to specify a region, as restic does a region lookup on your bucket.

**Comment by trondhindenes** [comment]
Thanks!

Hm, maybe I can sneak the correct s3 endpoint in there using something like `s3.eu-west-1.amazonaws.com/<bucketname>` in the config. Looks like restic itself just expects a parseable s3-comptatible hostname (looking at their docs). I'll try that tomorrow.

**Comment by trondhindenes** [comment]
oh, if restic does region lookups it probably needs a full ListBuckets:* permission, which isn't the case if I just duplicate the ark s3 setup.

**Comment by ncdc** [comment]
From what I recall, region lookups don’t use credentials and don’t require
permissions.

**Comment by trondhindenes** [comment]
Everything worked fine after adding:
```
{
            "Effect": "Allow",
            "Action": [
                "s3:GetBucketLocation",
                "s3:ListAllMyBuckets"
            ],
            "Resource": [
                "arn:aws:s3:::*"
            ]
        }
```
(not sure both are needed)

**Comment by trondhindenes** [comment]
yup, the header is there.
